### PR TITLE
Support doorbell motion zone (de)activation

### PR DESCRIPTION
**Pull request by arkpog**
Add method for activating/deactivating/deleting doorbell motion zones.

### CODE REVIEW

File: ring_doorbell/doorbot.py [hunk 1]
```diff
@@ -340,6 +341,33 @@ def volume(self, value):
...
 
+    @property
+    def motion_zones(self):
+        return self._attrs.get('settings').get('motion_zones')
+
+    def motion_zone_state(self, zone_id, value):
+        """Set a motion zone to one of the allowed states. See
+        `const.DOORBELL_MOTION_ZONE_STATE`"""
+
+        if not zone_id in DOORBELL_MOTION_ZONES:
```
[the remark this answers is not in the document]
Comment (tchellomello) [reply ring_doorbell/doorbot.py]: @arkpog  could you please re-write this conditional statement?

File: ring_doorbell/doorbot.py [hunk 2]
```diff
@@ -340,6 +341,33 @@ def volume(self, value):
...
+            return False
+
+        if value not in DOORBELL_MOTION_ZONE_STATE:
+            _LOGGER.error(
+                MSG_ALLOWED_VALUES.format(list(DOORBELL_MOTION_ZONE_STATE)))
+            return False
+
+        params = {
+            'doorbot[description]': self.name,
+            'doorbot[settings][motion_zones][{0}][state]'.format(zone_id): value
```
[the remark this answers is not in the document]
Comment (tchellomello) [reply ring_doorbell/doorbot.py]: @arkpog please keep it within 79 chars

[the remark this answers is not in the document]
Comment (arkpog) [reply ring_doorbell/doorbot.py]:
```suggestion
            'doorbot[settings][motion_zones][{0}][state]'.format(zone_id):\
                value
```

File: ring_doorbell/doorbot.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arkpog) [reply ring_doorbell/doorbot.py]:
```suggestion
        if zone_id not in DOORBELL_MOTION_ZONES:
```

**Comment by FearNaBoinne** [comment]
What's the state of this request? It looks like it was abandoned Feb this year, but it would be great to be able to control the zones, so I can link them to my Domoticz system and enable/disable zones together with the rest of the alarm system...

**Comment by FearNaBoinne** [comment]
Also, is there something I can do to help move this forward?

**Review by tchellomello — CHANGES_REQUESTED** [review]
@arkpog sorry for the long delay in reviewing this. I'm back on looking at this project more closely.
Could you re-work your PR so we can get this merged? Basically lint issues and re-testing it against the latest master branch so we can validate the change? Thanks

**Comment by jfmennedy** [comment]
Tried this but getting back:
`requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: https://api.ring.com/clients_api/doorbots/5xxx9xxx?api_version=9&doorbot%5Bdescription%5D=Haustuer&doorbot%5Bsettings%5D%5Bmotion_zones%5D%5Bzone2%5D%5Bstate%5D=0
`
Is there a chance to get this to work? Or even to pause motiondetection for a given period (30min,1h,2h,3h,4h) as in the app?

Thx Jan

File: ring_doorbell/doorbot.py [hunk 3]
```diff
@@ -360,8 +360,8 @@ def motion_zone_state(self, zone_id, value):
 
         params = {
             'doorbot[description]': self.name,
-            'doorbot[settings][motion_zones][{0}][state]'.format(zone_id): value
-        }
+            'doorbot[settings][motion_zones][{0}][state]'.format(zone_id):
+                value}
 
         url = API_URI + DOORBELLS_ENDPOINT.format(self.account_id)
```
Comment (jfmennedy) [inline ring_doorbell/doorbot.py:368]: url = DOORBELLS_ENDPOINT.format(self.account_id)

Comment (jfmennedy) [inline ring_doorbell/doorbot.py:368]: url = DOORBELLS_ENDPOINT.format(self.account_id)

**Pull request closed without merging**